doc/src/modules/concrete.rst

```diff
@@ -5,23 +5,20 @@ Concrete
 Hypergeometric terms
 --------------------
 
-The center stage, in recurrence solving and summations, play hypergeometric
-terms. Formally these are sequences annihilated by first order linear
-recurrence operators. In simple words if we are given term `a(n)` then it is
-hypergeometric if its consecutive term ratio is a rational function in `n`.
+Hypergeometric terms play a prominent role in the theory of recurrences and
+summation. A sequence `a(n)` is a hypergeometric term provided that the
+consecutive term ratio `a(n + 1) / a(n)` is a rational function in `n`. Such
+sequences are also called just "hypergeometric." Hypergeometric terms include
+polynomials, rational functions, factorials, exponential functions, and
+binomial coefficients, along with products and quotients of any of these.
 
-To check if a sequence is of this type you can use the ``is_hypergeometric``
-method which is available in Basic class. Here is simple example involving a
-polynomial:
+The ``is_hypergeometric`` method checks if an expression is hypergeometric.
+Here are some examples:
 
     >>> from sympy import *
     >>> n, k = symbols('n,k')
     >>> (n**2 + 1).is_hypergeometric(n)
     True
-
-Of course polynomials are hypergeometric but are there any more complicated
-sequences of this type? Here are some trivial examples:
-
     >>> factorial(n).is_hypergeometric(n)
     True
     >>> binomial(n, k).is_hypergeometric(n)
```

Comment:
I like the chatter in the docstrings which act like a guide as you read. Perhaps
```python
While polynomials are hypergeometric, other functions often encountered
in work with series are as well:

    >>> (2**n).is_hypergeometric(n)  # powers
    >>> all(binomial(n, k).is_hypergeometric(a) for a in (n, k))  # binomials
    True
    >>> all(_f(n, k).is_binomial(a) for _f in (rf, ff) for a in (n, k))  # rising and falling factorials
    True

    Compositions of hypergeometric terms are usually not hypergeometric:

    >>> factorial(n**2).is_hypergeometric(n)
    False
    >>> (2**(n**3 + 1)).is_hypergeometric(n)
    False
```

Reply:
I'm happy to have more chatter. Module "splash pages" seem like a good place to do this.